app/trans/http/user.go

```diff
@@ -0,0 +1,21 @@
+package http
+
+import (
+	"net/http"
+)
+
+// addUser will create User
+func (ah *ApiHandler) addUser(rw http.ResponseWriter, req *http.Request) {
```

Comment:
I'm confused a bit, cos we already have this func - "func (ah *ApiHandler) HandleUser() http.Handler {".
I thought that was a handler. But now I see a handler here.
So, for what do we need that first func?

Reply:
After addressing some naming related problems code may look like this:
```go
// Handle will take care of our handsome User's endpoint
func (uh UserHandler) Handle() http.Handler {
	return http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
		switch req.Method {
		case http.MethodPost:
			uh.add(rw, req)
		default:
		}
	})
}

// add will create User
func (uh UserHandler) add(rw http.ResponseWriter, req *http.Request) {
```

Reply:
Another pice of the puzzle. Not sure it would work properly without pointer receivers
```go
func (m Mux) mapRoutes(uh UserHandler) {
	m.Handle("/user", uh.Handle()) // + middleware
}
```